app/services/declarations/create.rb

```diff
@@ -0,0 +1,162 @@
+# frozen_string_literal: true
+
+module Declarations
+  class Create
+    include ActiveModel::Model
+    include ActiveModel::Attributes
+
+    attribute :lead_provider
+    attribute :participant_id
+    attribute :declaration_type, :string
+    attribute :declaration_date, :datetime
+    attribute :course_identifier, :string
+    attribute :has_passed
+
+    validates :lead_provider, presence: true
+    validates :participant_id, presence: true
+    validates :participant, participant_presence: true, participant_not_withdrawn: true
+    validates :course_identifier, inclusion: { in: Course::IDENTIFIERS }, allow_blank: false, course_for_participant: true
+    validates :declaration_date, declaration_date: true
+    validates :declaration_date, presence: true
+    validates :declaration_type, presence: true
+    validates :cohort, contract_for_cohort_and_course: true
+
+    validate :output_fee_statement_available
+    validate :validate_has_passed_field, if: :validate_has_passed?
+    validate :validate_schedule_exists
+    validate :validates_billable_slot_available
+
+    attr_reader :raw_declaration_date
+
+    def save
```

Comment:
To maintain consistency with other service classes, can we rename `save` to `create`

Reply:
rubcop 🤦